src/main/java/org/hisp/dhis/antlr/AntlrParserUtils.java

```diff
@@ -280,6 +280,10 @@ private static Object castDoubleObject( Double object, Class<?> clazz )
         {
             return object.toString();
         }
+        else if ( clazz == Boolean.class && object % 1 == 0 )
+        {
+            return Boolean.parseBoolean( object.toString() );
```

Comment:
We should return something like this to implement the logic that 0 is false and any other number is true
```suggestion
            return object != 0.0;
```

The Boolean.parseBoolean not considering numbers as booleans, so any number would be false

Reply:
make sense